meta/saiserialize.c

```diff
@@ -1243,6 +1382,77 @@ int sai_deserialize_attribute(
         _In_ const char *buffer,
         _Out_ sai_attribute_t *attribute)
 {
-    SAI_META_LOG_WARN("not implemented");
-    return SAI_SERIALIZE_ERROR;
+    const char *buf = buffer;
+    const sai_attr_metadata_t *meta;
+    bool deserialize_binary = false;
+    size_t len;
+    int ret;
+    uint32_t idx;
+
+    EXPECT("{");
+
+    EXPECT_KEY("id");
+
+    EXPECT("\"");
+
+    meta = sai_metadata_get_attr_metadata_by_attr_id_name(buf);
+    if (NULL!= meta)
+    {
+        len = strlen(meta->attridname);
+        attribute->id = meta->attrid;
+        buf += len;
+        SAI_META_LOG_WARN("id: %u, name: %s", meta->attrid, meta->attridname);
+    }
+    else
+    {
+        EXPECT_CHECK(sai_deserialize_uint32(buf, &attribute->id), uint32);
+        SAI_META_LOG_WARN("id: %u, name not found", attribute->id);
+    }
+
+    EXPECT("\"");
+
+    EXPECT_NEXT_KEY("value");
+
+    if (NULL != meta)
```

Comment:
please move NULL to the right of compare, keep the same code quality as rest of the file

Reply:
ACK